content/en/blog/_posts/2022-11-28-advancements-in-traffic-engineering/index.md

```diff
@@ -0,0 +1,75 @@
+
+In Kubernetes, you can create a Service with Type=LoadBalancer to expose an application externally with a load balancer.
+The load balancer implementation varies between clusters and platforms, but the Service provides a generic abstraction
+representing the load balancer that is consistent across all Kubernetes installations.
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: my-service
+spec:
+  selector:
+    app.kubernetes.io/name: my-app
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 9376
+  type: LoadBalancer
+```
+
+Under the hood, Kubernetes allocates a NodePort for the Service, which is then used by kube-proxy to provide a
+network data path from the NodePort to the Pod. A controller running in the Kubernetes Control Plane will then
+add all available Nodes in the cluster to the load balancer’s backend pool, using the designated NodePort for
+the Service as the backend target port.
+
+{{< figure src="traffic-engineering-service-load-balancer.png" alt="Figure 1: Overview of Service Load Balancers" >}}
+
+Oftentimes it is beneficial to set `externalTrafficPolicy: Local` for Services, to avoid extra hops between
+Nodes that are not running healthy Pods backing that Service. When using `externalTrafficPolicy: Local`,
+an additional NodePort is allocated for health checking purposes, such that Nodes that do not contain healthy
+Pods are not included in the backend pool for a load balancer.
+
+{{< figure src="traffic-engineering-lb-healthy.png" alt="" >}}
+
+One such scenario where traffic can be lost is when a Node loses the last remaining Pod for a Service,
+but the external load balancer has not probed the health check NodePort yet. The likelihood of this situation
+is largely dependent on the health checking interval configured on the load balancer. The larger the interval,
+the more likely this will happen, since the load balancer will take longer to realize that a node is no longer
+healthy and unable to accept traffic. This also occurred when Pods started terminating during rolling updates,
+specifically when a Node went from N to 0 healthy pods because Kubernetes does not consider terminating Pods
+as “Ready”, regardless of their readiness probes.
+
+{{< figure src="traffic-engineering-lb-unhealthy.png" alt="" >}}
+
+Starting in Kubernetes v1.26, kube-proxy enables the `ProxyTerminatingEndpoints` feature by default, which
+adds automatic failover and routing to terminating endpoints in scenarios where the traffic would otherwise
+be dropped. More specifically, when there is a rolling update and a Node only contains terminating replicas,
+kube-proxy will route traffic to the terminating replicas as long as their readiness probes are passing.
+By doing so, kube-proxy provides the external load balancer a window of time to gracefully steer traffic
+away from the Node after its next health check probe.
```

Comment:
Is this right:
```suggestion
Starting in Kubernetes v1.26, kube-proxy defaults to using _proxy terminating endpoints_ behavior. The
kube-proxy spots when traffic arrives that is destined to a terminating endpoint, and avoids some scenarios where the traffic would otherwise be dropped.

More specifically, when there is a rolling update and a Node only contains terminating replicas,
kube-proxy routes traffic to the terminating replicas only as long as their readiness probes are passing.
By doing so, kube-proxy provides the external load balancer a window of time to gracefully steer traffic
away from the Node after its next health check probe.
If there are no local healthy replicas, kube-proxy looks for another node with healthy endpoints and
tries to send the traffic there.
```
?

I'm worried that readers won't work out how the new behavior is different.